src/quota.rs

```diff
@@ -0,0 +1,201 @@
+use anyhow::{anyhow, bail, Result};
+use async_imap::types::{Quota, QuotaResource};
+use humansize::{file_size_opts, FileSize};
+use itertools::Itertools;
+
+use crate::context::Context;
+use crate::imap::Imap;
+use crate::{
+    chat::{add_device_msg, add_device_msg_with_importance},
+    constants::Viewtype,
+    imap::scan_folders::get_watched_folders,
+    message::Message,
+};
+use crate::{
+    constants::DC_QUOTA_WARN_THRESHOLD_PERCENTAGE,
+    stock_str::{
+        quota_mailbox_nearly_full, quota_not_supported, quota_resource_messages,
+        quota_resource_storage,
+    },
+};
+
+/// Generates a detailed report about the current Quota usage on the for deltachat relevant folders
+/// and sends it to the user via [add_device_msg]
+///
+/// It's a bit like the prepaid mobile carrier service menu/messages,
+/// where you type a special number and then get a message back with your current balance.
+pub(crate) async fn quota_usage_report_job(context: &Context, imap: &mut Imap) -> Result<()> {
+    if !imap.capabilities_determined() {
+        bail!("imap is not ready");
+    }
+
+    if imap.can_check_quota() {
+        let folders = get_watched_folders(context).await;
+        let unique_quota_roots = get_unique_quota_roots_and_usage(folders, imap).await?;
+
+        // build and send report message
+        let mut msg = Message::new(Viewtype::Text);
+        msg.text = Some(generate_report_message(&unique_quota_roots, context).await?);
+        add_device_msg(context, None, Some(&mut msg)).await?;
+    } else {
+        warn!(
+            context,
+            "the email server does not support the quota extention"
+        );
+        let mut msg = Message::new(Viewtype::Text);
+        msg.text = Some(quota_not_supported(context).await); // todo stock string?
+        add_device_msg_with_importance(context, None, Some(&mut msg), true).await?;
+    }
+
+    Ok(())
+}
+
+async fn generate_report_message(
+    unique_quota_roots: &[(String, Vec<QuotaResource>)],
+    context: &Context,
+) -> Result<String> {
+    let mut message: String = "".to_owned();
+
+    let storage_stock_string = quota_resource_storage(context).await;
+    let messages_stock_string = quota_resource_messages(context).await;
+    for (name, quota_resources) in unique_quota_roots {
+        message.push_str(&format!("{}:\n", &name));
+        use async_imap::types::QuotaResourceName::*;
+        for resource in quota_resources {
+            message.push_str(&match &resource.name {
+                Atom(name) => {
+                    format!("[{}/{}] {}\n", resource.usage, resource.limit, name)
+                }
+                Message => {
+                    format!(
+                        "{}/{} {}\n",
```

Comment:
I would prefer "{}: {}/{}" here and for `Atom` and for `Storage` too, otherwise it is hard to translate as you can't simply use nominative case.